### PR TITLE
Introduce the new user-facing Logs API as a placeholder

### PR DESCRIPTION
**Pull request by MSNev**
Part of https://github.com/open-telemetry/oteps/pull/265

### CODE REVIEW

File: specification/README.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ path_base_for_github_subdir:
   - [Tracing](trace/api.md)
   - [Metrics](metrics/api.md)
   - [Logs](logs/README.md)
+    - [Public API](logs/public-api.md)
```
Comment (lmolkova) [inline specification/README.md:32]: having a new API comes with a cost:

- cognitive overload due to multiple concepts representing a similar thing that ends up on the same exporter in the same format
- configuration complexity for 3 SDK pipelines

So it has clear drawbacks, but I don't see any problem it solves or benefits it brings.

PS: 2 of these concepts were enough to spend weeks bikeshedding, having 3 (3!!) concepts is not going to help us make progress.

File: specification/logs/public-api.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Public Logs API
```
Comment (tedsuo) [inline specification/logs/public-api.md:1]: fyi there's no need to use the word "public" in the spec, the term API is defined to be public.

File: specification/logs/bridge-api.md [hunk 3]
```diff
@@ -44,6 +45,14 @@ graph TD
     B -->|Emit| C(LogRecord)
 ```
 
+## Public API Development
```
Comment (trask) [inline specification/logs/bridge-api.md:48]: (similar to Enabled which is in status development)

```suggestion
## Instrumentation API

Status: Development
```

File: specification/logs/event-api.md [hunk 4]
```diff
@@ -30,6 +31,16 @@ The Event API consists of these main components:
   provides access to `EventLogger`s.
 * [EventLogger](#eventlogger) is the component responsible for emitting events.
 
+## Instrumentation API Development
+
+**Status**: [Development](../document-status.md)
```
Comment (pellared) [inline specification/logs/event-api.md:36]: nit: Not sure if this is needed. The whole document is "development"

File: specification/README.md (hunk 1 above)
Comment (MSNev) [reply specification/README.md]: Discussed in the Logs and Spec SIG and removed and folded into the existing bridge-api file.

File: specification/logs/event-sdk.md [hunk 5]
```diff
@@ -29,6 +30,14 @@ API that provides users with this functionally.
 
 All implementations of the OpenTelemetry API MUST provide an SDK.
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
+
+The intent is that this new API will incorporate the current functionality of this existing API and once it is defined and implemented, the Event API usage will be migrated, deprecated, renamed and eventually removed.
+
+No further work is scheduled for the current Event SDK at this time.
```
Comment (pellared) [inline specification/logs/event-sdk.md:39]:
```suggestion
The intent is that Logs SDK will incorporate the current functionality of this existing SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.

No further work is scheduled for the current Events SDK at this time.
```

Comment (MSNev) [reply specification/logs/event-sdk.md:39]: Let me adjust this in all references.

Comment (MSNev) [reply specification/logs/event-sdk.md:39]: I'm going to call it Instrumentation API for consistency though.

File: specification/logs/event-api.md [hunk 6]
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
   provides access to `EventLogger`s.
 * [EventLogger](#eventlogger) is the component responsible for emitting events.
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
+
+The intent is that this Instrumentation API will incorporate the current functionality of this existing API and once it is defined and implemented, the Event API usage will be migrated, deprecated, renamed and eventually removed.
```
Comment (pellared) [inline specification/logs/event-api.md:38]: nit

```suggestion
The intent is that this Instrumentation API will incorporate the current functionality of this existing API and once it is defined and implemented, the Events API usage will be migrated, deprecated, renamed and eventually removed.
```

File: specification/logs/event-api.md [hunk 7]
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
...
+
+No further work is scheduled for the current Event API definition at this time.
```
Comment (pellared) [inline specification/logs/event-api.md:40]:
```suggestion
No further work is scheduled for the current Events API definition at this time.
```

File: specification/logs/bridge-api.md [hunk 8]
```diff
@@ -145,6 +146,16 @@ SHOULD be documented that instrumentation authors needs to call this API each
...
 up-to-date response.
 
+## Instrumentation API
+
+**Status**: [Development](../document-status.md)
+
+This set of API functions will provide the capabalities needed to emit a
+`LogRecord` as is currently provided by
+
+* [Logs Bridge API](./bridge-api.md)
```
Comment (pellared) [inline specification/logs/bridge-api.md:156]:
```suggestion
```

I think we agreed that for now we that the API just needs to be capable of emitting events (log records with `event.name` attribute).

File: specification/logs/bridge-api.md [hunk 9]
```diff
@@ -145,6 +146,15 @@ SHOULD be documented that instrumentation authors needs to call this API each
...
 up-to-date response.
 
+## Instrumentation API
+
+**Status**: [Development](../document-status.md)
+
+This set of API functions will provide the capabalities needed to emit a
+`LogRecord` as is currently provided by
+
+* [Events API](./event-api.md)
```
Comment (pellared) [inline specification/logs/bridge-api.md:156]:
```suggestion
`LogRecord` as is currently provided by [Events API](./event-api.md).
```

**Review by pellared — APPROVED** [review]

File: specification/logs/bridge-api.md [hunk 10]
```diff
@@ -145,6 +146,13 @@ SHOULD be documented that instrumentation authors needs to call this API each
 time they [emit a LogRecord](#emit-a-logrecord) to ensure they have the most
 up-to-date response.
 
+## Instrumentation API
+
+**Status**: [Development](../document-status.md)
+
+This set of API functions will provide the capabalities needed to emit a
```
Comment (trask) [inline specification/logs/bridge-api.md:153]:
```suggestion
This set of API functions will provide the capabilities needed to emit a
```

File: specification/logs/event-api.md [hunk 11]
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
   provides access to `EventLogger`s.
 * [EventLogger](#eventlogger) is the component responsible for emitting events.
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
```
Comment (trask) [inline specification/logs/event-api.md:36]:
```suggestion
Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

File: specification/logs/event-api.md [hunk 12]
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
...
+
+The intent is that this Instrumentation API will incorporate the current functionality of this existing API and once it is defined and implemented, the Events API usage will be migrated, deprecated, renamed and eventually removed.
```
Comment (trask) [inline specification/logs/event-api.md:38]:
```suggestion
The intent is that this Logs Instrumentation API will incorporate the current functionality of this existing Events API and once it is defined and implemented, the Events API usage will be migrated, deprecated, renamed and eventually removed.
```

File: specification/logs/event-sdk.md [hunk 13]
```diff
@@ -29,6 +30,14 @@ API that provides users with this functionally.
 
 All implementations of the OpenTelemetry API MUST provide an SDK.
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
```
Comment (trask) [inline specification/logs/event-sdk.md:35]:
```suggestion
Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

File: specification/logs/event-sdk.md [hunk 14]
```diff
@@ -29,6 +30,14 @@ API that provides users with this functionally.
...
+
+The intent is that Logs SDK will incorporate the current functionality of this existing SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.
```
Comment (trask) [inline specification/logs/event-sdk.md:37]:
```suggestion
The intent is that Logs SDK will incorporate the current functionality of this existing Events SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.
```

File: specification/logs/sdk.md [hunk 15]
```diff
@@ -538,4 +539,8 @@ return a Failure result.
 and the destination is unavailable). [OpenTelemetry SDK](../overview.md#sdk)
 authors MAY decide if they want to make the shutdown timeout configurable.
 
+## Instrumentation API
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
```
Comment (trask) [inline specification/logs/sdk.md:544]:
```suggestion
Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

**Review by trask — APPROVED** [review]
(editorial suggestions only)

**Review by pellared — APPROVED** [review]

**Review by trask — APPROVED** [review]

File: specification/logs/event-api.md [hunk 16]
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
   provides access to `EventLogger`s.
 * [EventLogger](#eventlogger) is the component responsible for emitting events.
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```
Comment (lmolkova) [inline specification/logs/event-api.md:36]: super-nit
```suggestion
> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

will be rendered as 

> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).

File: specification/logs/event-sdk.md [hunk 17]
```diff
@@ -29,6 +30,14 @@ API that provides users with this functionally.
 
 All implementations of the OpenTelemetry API MUST provide an SDK.
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```
Comment (lmolkova) [inline specification/logs/event-sdk.md:35]:
```suggestion
> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

File: specification/logs/sdk.md [hunk 18]
```diff
@@ -538,4 +539,8 @@ return a Failure result.
 and the destination is unavailable). [OpenTelemetry SDK](../overview.md#sdk)
 authors MAY decide if they want to make the shutdown timeout configurable.
 
+## Instrumentation API
+
+Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```
Comment (lmolkova) [inline specification/logs/sdk.md:544]:
```suggestion
> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

**Review by lmolkova — APPROVED** [review]

**Review by tedsuo — APPROVED** [review]
Looks great, let's get this merged and add the API method.

**Comment by MSNev** [comment]
@trask, @lmolkova, @pellared can you please check the latest commit, it (should) include
- all of your suggested changes (which I committed from github)
- As per discussion in the Maintainers meeting (general agreement of calling it the "Logs Instrumentation API"), updated the missed renames as per Trasks commits from last week.

Process taken: Merged changes from remote branch (which included the commits), updated missed renamings and pushed them back out (leading to loads of listed "commits" -- which should just be duplicative), I normally "cleanup" and squash the commits locally to keep it clean, but that requires a forced push to my branch and I didn't want to do that for this PR, hence the "messy" looking commits.

File: specification/logs/event-sdk.md [hunk 19]
```diff
@@ -29,6 +30,15 @@ API that provides users with this functionally.
 
 All implementations of the OpenTelemetry API MUST provide an SDK.
 
+## Logs Instrumentation API Development
+
+> [!NOTE]
+> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#logs-instrumentation-api).
+
+The intent is that Logs SDK will incorporate the current functionality of this existing Events SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.
```
[the remark this answers is not in the document]
Comment (MSNev) [reply specification/logs/event-sdk.md:38]: This note is for anyone reading it between now and when we have the new Logs API functionality specified.

I am aware that there is active work occurring (as this has existed for several years) and there are some instrumentations that are using this today, I believe in JS, Java (Android) and potentially now in Python (for LLM).

I'm also trying to explicitly defer the exact approach that we take until we get to that point, so by using these enumerations it (hopefully) is also acting as a further warning to not take any dependency on it (if they have not already) and for anyone how has we will hopefully get feedback.

**Review by pellared — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Review by mattmccleary — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**